opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -817,6 +824,24 @@ namespace Opm
             return solvent_well_rate;
         }
 
+        /// One rate pr well connection.
+        std::vector<double>& perfRatePolymer() { return perfRatePolymer_; }
+        const std::vector<double>& perfRatePolymer() const { return perfRatePolymer_; }
+
+        /// One rate pr well
+        double polymerWellRate(const int w) const {
+            int connpos = 0;
+            for (int iw = 0; iw < w; ++iw) {
+                connpos += this->well_perf_data_[iw].size();
+            }
```

Comment:
If we're going to do this a lot there should really be a precomputed lookup table for the start of the perforations of each well.

Reply:
There is already a first_perf_index_ vector in BlackoilWellModel that passes the corresponding first_perf to the WellInterface. It is only (I think) used to access the perforation data from the wellState.  Maybe we should make a member in the wellState instead?

Reply:
>> If we're going to do this a lot there should really be a precomputed lookup table for the start of the perforations of each well.
>
> Maybe we should make [`BlackoilWellModel::first_perf_index_`] a member in the wellState instead?

If that's feasible then I think that would be a good idea, at least for as long as the well state objects maintains the states of all wells as opposed to the state of a single well (cf. #2582).

Reply:
I have kept the first_perf in wellInteface, but moved it from BlackoilWellModel to the BlackoilWellState.